daemon/compress.c

```diff
@@ -0,0 +1,75 @@
+
+#include <str.h>
+#include <zlib.h>
+#include <string.h>
+#include <errno.h>
+
+#include "log.h"
+#include "udp_listener.h"
+#include "str.h"
+#include "compress.h"
+
+#define windowBits 15
+
+#define GZIP_ENCODING 16
+#define ENABLE_ZLIB_GZIP 32
+
+static void init_stream(z_stream *stream);
+
+static void init_stream(z_stream *stream)
+{
+	stream->zalloc = Z_NULL;
+	stream->zfree = Z_NULL;
+	stream->opaque = Z_NULL;
+}
+
+
+// compress_data compress the data of the given length.
+// returns compressed buffer length and replace the original buffer.
+int compress_data(char *buf, int len) {
+	uLong comp_len = compressBound(len);
+    char tmp[comp_len];
```

Comment:
I'm not sure what's @rfuchs take on VLAs?

Reply:
What is the VLAs?

Reply:
A VLA is a variable-length array.

Reply:
Thanks, now I've got that. Btw, it's gone now. :) I'm resolving this.